### PR TITLE
All time widget configuration setup

### PR DESCRIPTION
**Pull request by planarvoid**
Since Views widget and All-time widget have the same configuration, I'm reusing the configuration fragment and viewmodel. The only difference is the title (which comes from the separate configuration activities) and the action to update the widget. Both the color and the selected site are stored per `appWidgetId` which is unique. The All-time widget has a different placeholder icon.

The widget updated class is not yet implemented (it's copy/paste of the views updater). I'll implement that in the next PR.

To test:
* Add a widget 
* Notice there is new widget with a different placeholder icon
* Place it on the homescreen
* The configuration opens
* The title is "All-time"
* The color and site selector work as expected

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/StatsWidgetConfigureFragment.kt [hunk 1]
```diff
@@ -12,22 +14,49 @@ import androidx.lifecycle.Observer
...
+    @Inject lateinit var appPrefsWrapper: AppPrefsWrapper
+    @Inject lateinit var siteStore: SiteStore
+    @Inject lateinit var imageManager: ImageManager
+    private lateinit var viewModel: StatsWidgetConfigureViewModel
+    private lateinit var viewType: ViewType
+
+    override fun onInflate(context: Context?, attrs: AttributeSet?, savedInstanceState: Bundle?) {
+        super.onInflate(context, attrs, savedInstanceState)
+        activity?.let {
+            val a = it.obtainStyledAttributes(attrs, R.styleable.statsWidget)
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/StatsWidgetConfigureFragment.kt]: Maybe we could use a bit more descriptive variable name here :)

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/StatsWidgetConfigureFragment.kt [hunk 2]
```diff
@@ -12,22 +14,49 @@ import androidx.lifecycle.Observer
...
+            val views = a.getInt(R.styleable.statsWidget_viewType, -1)
+            viewType = when (views) {
+                -1 -> throw IllegalArgumentException("The view type needs to be specified on the fragment")
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/StatsWidgetConfigureFragment.kt]: I assume -1 case will be covered by else? If so maybe we can remove it :)

File: WordPress/src/main/res/layout/stats_all_time_widget_configure_activity.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <androidx.coordinatorlayout.widget.CoordinatorLayout
+        android:id="@+id/coordinatorLayout"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <fragment
+            android:id="@+id/fragment_container"
```
Comment (khaykov) [inline WordPress/src/main/res/layout/stats_all_time_widget_configure_activity.xml]: This is pretty minor, but maybe we could stick to a single naming style within layout file (camelCase?)

**Review by khaykov — COMMENTED** [review]
Thanks for the PR, @planarvoid ! Everything looks good, but I have a couple of super minor comments about code :)

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/StatsWidgetConfigureFragment.kt (hunk 2 above)
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/StatsWidgetConfigureFragment.kt]: you're right 👍

File: WordPress/src/main/res/layout/stats_all_time_widget_configure_activity.xml (hunk 3 above)
Comment (planarvoid) [reply WordPress/src/main/res/layout/stats_all_time_widget_configure_activity.xml]: I haven't noticed the camel case :-), this class is a copy&paster and I was just adjusting the content and the toolbar. I think IDs are usually snake case, right? I'm ok with both 👍 . I've changed the camel case to snake case in the parent layout

**Review by khaykov — APPROVED** [review]
Thanks for the changes! Looks good 👍

**Pull request merged**